### DISCUSSION TITLE
Is it possible to choose which region(s) the Edge functions run in?

### DISCUSSION THREAD

**Discussion by jsonMartin** [supabase/supabase#12613] [Questions]
Hello, I've been searching for an answer to this question but can't find a definitive answer.

Is it possible to choose which region(s) the Edge functions run in? For example, Vercel allows this with their Serverless & Edge functions. It also appears Deno can specify [endpoint URL](https://deno.com/subhosting/docs/endpoints-and-regions) to target specific regions. 

My serverless functions query external APIs, then writes the results to Supabase table. The only available Region option I see within Supabase shows **Earth**, which means the function will always run closest to the user, not the Database:
<img width="358" alt="image" src="https://user-images.githubusercontent.com/4033621/220772850-8d1131b9-8e1f-41c5-a803-2c34bc07b891.png">

Ideally, my functions should all run in the same region as the Database, not the user! If the DB is hosted in US-West for example, but the User is from the EU, the edge function looks like: 
**EU Edge *(Query API)* -> US *(API Response)* -> EU Edge *(Invoke Supabase)* -> US *(Supabase Insert)* -> EU Edge *(Response)***

This takes longer because both the API Response & Supabase Insert needs to travel back and forth between US <-> EU. If Supabase Edge functions can specify runtime region, all of the API fetching & Supabase queries take place within the same region, reducing latency. 

For my use case, Serverless is one of the major selling points of using Supabase Hosted offering & Pro plan. I'd rather do this all within Supabase & Deno if possible, rather than using Vercel Serverless or using a NodeJS backend server located in the same region as Supabase, but latency is an important factor to consider.

**Comment by laktek** [comment]
@jsonMartin We are working on providing the option to configure the region(s) to serve edge functions. Will update this thread when we ship it (ETA: 4-8 weeks).

**Comment by jsonMartin** [comment]
Awesome, thanks! Looking forward to this feature when it launches.

**Comment by joeldrotleff** [comment]
@laktek Any update on this?

I can confirm that having your DB and edge function in different regions seems to completely defeat the point of having edge functions. 

My 'hello-world' edge function gets great 100ms latency from any regions...

But my 'make-a-few-db-calls-and-return' function goes from 250ms in us-west to 1.1s in us-east to 2.3s in se-asia!

**Reply by laktek** [reply]
@joeldrotleff This feature got delayed since we had to solve some other priorities in Edge Function infrastructure. I'll prioritize it internally. Hoping to have it ready before the next Launch Week.

**Comment by laktek** [comment]
@jsonMartin @joeldrotleff We just shipped the option to select the region when invoking edge functions.

Here are the docs on how to use it https://supabase.com/docs/guides/functions/regional-invocation

**Reply by jsonMartin** [reply]
> @jsonMartin @joeldrotleff We just shipped the option to select the region when invoking edge functions.
> 
> 
> 
> Here are the docs on how to use it https://supabase.com/docs/guides/functions/regional-invocation

Awesome, glad to see this feature finally shipped. Thank you!

**Reply by tazmaniax** [reply]
@laktek is it planned to be able to specify the region in the `config.toml` like `verify_jwt`, e.g.
```
[functions.xapi]
verify_jwt = false
region = "eu-west-2"
```
